src/Operator/Constructors.jl

```diff
@@ -780,6 +781,184 @@ function advectionoperator(basis::AbstractBasis, mesh::Mesh, wind = [1, 1])
+
+2 inputs:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation modes:
+    interpolation
+    gradient
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    quadratureweights
+
+1 output:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    gradient
+```
+"""
+P = 2
+τ = 0.5 / (P - 1) # Tau scaling for SUPG, 0 returns Galerkin method
+function supgadvectionoperator(basis::AbstractBasis, mesh::Mesh, wind = [1, 1])
+    function supgadvectionweakform(U::Matrix{Float64}, w::Array{Float64})
+        u = U[1, :]
+        du = U[2, :]
+        dv = (wind * u - wind * τ * (wind * du)) * w[1]
+        return [dv]
+    end
+
+    # fields
+    inputs = [
+        OperatorField(basis, [EvaluationMode.interpolation, EvaluationMode.gradient]),
+        OperatorField(basis, [EvaluationMode.quadratureweights]),
+    ]
+    outputs = [OperatorField(basis, [EvaluationMode.gradient])]
+
+    # operator
+    supgadvection = Operator(supgadvectionweakform, mesh, inputs, outputs)
+    return supgadvection
+end
+
+"""
+```julia
+supgmassoperator(basis, mesh, wind)
+```
+Convenience constructor for SUPG mass matrix operator
+
+# Weak form: Left hand side
+- ``\\int v u_t + wind τ u_t \\nabla v``
+
+# Arguments:
+- `basis`: basis for all operator fields to use
+- `mesh`:  mesh for operator
+- `wind`:  advection speed in 2D
+- `τ`:     scaling for SUPG
+
+# Returns:
+- SUPG mass matrix operator with basis on mesh
+
+# Example:
+```jldoctest
+# supg mass matrix operator
+mesh = Mesh2D(1.0, 1.0);
+mapping = nothing
+#mapping = sausage_transformation(9);
+basis = TensorH1LagrangeBasis(3, 4, 1, mesh.dimension, collocatedquadrature = false, mapping = mapping)
+wind = [1, 1]
+supgmass = LFAToolkit.supgmassoperator(basis, mesh, wind);
+
+# verify
+println(supgmass)
+
+# output
+
+finite element operator:
+2d mesh:
+    dx: 1.0
+    dy: 1.0
+
+2 inputs:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    interpolation
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    quadratureweights
+
+1 output:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation modes:
+    interpolation
+    gradient
+```
+"""
+function supgmassoperator(basis::AbstractBasis, mesh::Mesh, wind = [1, 1])
+    function supgmassweakform(udot::Array{Float64}, w::Array{Float64})
+        v = udot * w[1]
+        dv = wind * τ * udot * w[1]
+        return ([v; dv],)
```

Comment:
@jeremylt This structure seems to be necessary due to how it's unpacked, but perhaps there that can be flattened/simplified without losing capability.

Reply:
This simplification should work, or is at least intended to work
```suggestion
        return [v; dv]
```